src/sparse/KokkosSparse_spgemm_jacobi.hpp

```diff
@@ -0,0 +1,249 @@
+		     typename clno_row_view_t_::non_const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Output matrix rowmap must be non-const.");
+
+      static_assert (std::is_same<typename clno_nnz_view_t_::value_type,
+		     typename clno_nnz_view_t_::non_const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Output matrix entriesView must be non-const.");
+
+      static_assert (std::is_same<typename cscalar_nnz_view_t_::value_type,
+		     typename cscalar_nnz_view_t_::non_const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Output matrix scalar view must be non-const.");
+
+      static_assert (std::is_same<typename KernelHandle::const_size_type,
+		     typename alno_row_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Size type of left handside matrix should be same as kernelHandle sizetype.");
+
+      static_assert (std::is_same<typename KernelHandle::const_size_type,
+		     typename blno_row_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Size type of right handside matrix should be same as kernelHandle sizetype.");
+
+      static_assert (std::is_same<typename KernelHandle::const_size_type,
+		     typename clno_row_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: Size type of output matrix should be same as kernelHandle sizetype.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_lno_t,
+		     typename alno_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: lno type of left handside matrix should be same as kernelHandle lno_t.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_lno_t,
+		     typename blno_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: lno type of right handside matrix should be same as kernelHandle lno_t.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_lno_t,
+		     typename clno_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: lno type of output matrix should be same as kernelHandle lno_t.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_scalar_t,
+		     typename ascalar_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: scalar type of left handside matrix should be same as kernelHandle scalar.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_scalar_t,
+		     typename bscalar_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: scalar type of right handside matrix should be same as kernelHandle scalar.");
+
+      static_assert (std::is_same<typename KernelHandle::const_nnz_scalar_t,
+		     typename cscalar_nnz_view_t_::const_value_type>::value,
+		     "KokkosSparse::spgemm_jacobi: scalar type of output matrix should be same as kernelHandle scalar.");
+
+
+      if (transposeA || transposeB){
+	throw std::runtime_error ("SpGEMM is not implemented for Transposes yet. "
```

Comment:
You mean spgemm-jacobi doesn't support transpose ..

Reply:
Done.